src/client.js

```diff
@@ -25,12 +25,22 @@ function History() {
     let historyLength = 0;
 
     function goBack() {
-        if (historyLength > 0) {
-            historyLength--;
-            history.goBack();
-        } else {
-            history.pushState(null, "/");
-        }
+        const currentUrl = location.toString();
+
+        historyLength--;
```

Comment:
What happens if the history length goes negative? Why even keep the length if we don't check it?